### PR TITLE
tts api changes

### PR DESCRIPTION
**Pull request by theomonnom**

### CODE REVIEW

File: livekit-agents/livekit/agents/ipc/proc_main.py [hunk 1]
```diff
@@ -209,13 +209,14 @@ def main(args: proto.ProcStartArgs) -> None:
...
         try:
-            tasks = asyncio.all_tasks(loop)
-            for task in tasks:
-                task.cancel()
-
-            loop.run_until_complete(asyncio.gather(*tasks, return_exceptions=True))
-            loop.run_until_complete(loop.shutdown_asyncgens())
-            loop.run_until_complete(loop.shutdown_default_executor())
-            loop.run_until_complete(cch.aclose())
+            pass
```
Comment (davidzhao) [inline livekit-agents/livekit/agents/ipc/proc_main.py:212]: eh?

File: livekit-plugins/livekit-plugins-azure/livekit/plugins/azure/tts.py [hunk 2]
```diff
@@ -158,6 +162,6 @@ def write(self, audio_buffer: memoryview) -> int:
             samples_per_channel=audio_buffer.nbytes // 2,
         )
 
-        audio = tts.SynthesizedAudio(text="", data=audio_frame)
+        audio = tts.SynthesizedAudio(segment_id=nanoid(), frame=audio_frame)
```
Comment (davidzhao) [inline livekit-plugins/livekit-plugins-azure/livekit/plugins/azure/tts.py:165]: does the single frame represent the entire segment? could we have it instantiate a new id if not already set? would be nice to have a default behavior

File: livekit-agents/livekit/agents/utils/misc.py [hunk 3]
```diff
@@ -43,3 +44,7 @@ def merge_frames(buffer: AudioBuffer) -> rtc.AudioFrame:
 
 def time_ms() -> int:
     return int(time.time() * 1000)
+
+
+def nanoid() -> str:
```
Comment (davidzhao) [inline livekit-agents/livekit/agents/utils/misc.py:49]: as a public API, it's not clear what `nanoid` is.. are you looking for a `shortuuid` ?

File: livekit-plugins/livekit-plugins-azure/livekit/plugins/azure/tts.py [hunk 4]
```diff
@@ -149,6 +152,7 @@ def __init__(
         super().__init__()
         self._event_queue = event_queue
         self._loop = loop
+        self._segment_id = nanoid()
```
Comment (davidzhao) [inline livekit-plugins/livekit-plugins-azure/livekit/plugins/azure/tts.py:155]: what is this used for?

File: livekit-plugins/livekit-plugins-azure/livekit/plugins/azure/tts.py (hunk 2 above)
Comment (theomonnom) [reply livekit-plugins/livekit-plugins-azure/livekit/plugins/azure/tts.py:165]: ignore this PR, I was too tired..

**Pull request merged**